core/src/main/scala/kafka/server/AbstractFetcherManager.scala

```diff
@@ -66,6 +68,15 @@ abstract class AbstractFetcherManager[T <: AbstractFetcherThread](val name: Stri
   Map("clientId" -> clientId)
   )
 
+  newGauge(
+    "FailedPartitionsCount", {
```

Comment:
Would be nice to have a test case for this metric. We can verify that it increments after a partition failure and that it decrements after a call to `removeFetcherForPartitions`.

Reply:
Added a test for the metric to verify its behavior.